libc/string_lenchr.c

```diff
@@ -0,0 +1,210 @@
+/*
+ * Phoenix-RTOS
+ *
+ * libc-tests
+ *
+ * libphoenix strlen and strnlen tests
+ *
+ * Copyright 2021, 2023 Phoenix Systems
+ * Author: Damian Modzelewski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+
+#include <stdio.h>
+#include <errno.h>
+#include <string.h>
+#include <limits.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <limits.h>
+
+#include <unity_fixture.h>
+
+
+TEST_GROUP(string_strlen);
+
+
+TEST_SETUP(string_strlen)
+{
+}
+
+
+TEST_TEAR_DOWN(string_strlen)
+{
+}
+
+
+TEST(string_strlen, strlen_right_input)
+{
+
+	const char empty[] = "\0";
+	char msg[] = { 'c', 'h', 'r', 'i', 's', 't', 'o', 'p', 'h', 'e', 'r', '\0' };
+	char *foo = "foo\0";
+	char *bar = "bar\0";
+	char str[12];
+
+	strcpy(str, "TEXT ");
+	strcat(str, foo);
+	strcat(str, bar);
+
+	TEST_ASSERT_EQUAL_INT(8, strlen("New Case"));
+	TEST_ASSERT_EQUAL_INT(0, strlen(empty));
+	TEST_ASSERT_EQUAL_INT(11, strlen(msg));
+	TEST_ASSERT_EQUAL_INT(11, strlen(str));
+}
+
+
+TEST(string_strlen, strlen_mix_input)
+{
+	/*Checking if some of input can change outcom*/
+	TEST_ASSERT_EQUAL_INT(16, strlen("Something \n Here"));
+	TEST_ASSERT_EQUAL_INT(16, strlen("Something \t Here"));
+	TEST_ASSERT_EQUAL_INT(16, strlen("Something \e Here"));
+	TEST_ASSERT_EQUAL_INT(16, strlen("Something \r Here"));
+	TEST_ASSERT_EQUAL_INT(16, strlen("Something \b Here"));
+	TEST_ASSERT_EQUAL_INT(16, strlen("Something \v Here"));
+	TEST_ASSERT_EQUAL_INT(16, strlen("Something \f Here"));
+	TEST_ASSERT_EQUAL_INT(16, strlen("Something \r Here"));
+	TEST_ASSERT_EQUAL_INT(25, strlen("Something Here \\ and here"));
+	TEST_ASSERT_EQUAL_INT(41, strlen("Something Here %d/%f/%x/%i/%c/%s and here"));
+	TEST_ASSERT_EQUAL_INT(0, strlen(""
+									""));
+	TEST_ASSERT_EQUAL_INT(51, strlen("12344567890!@#$^&*()_+~{[;'./<>?:/{empty}|:{}[]<>?:"));
+}
+
+
+TEST(string_strlen, strlen_special_input)
+{
+
+	char bigstr[5000] = "a";
+	/*Really big string to count*/
+	for (int i = 0; i < PATH_MAX - 1; i++) {
+		strcat(bigstr, "A");
+	}
+
+	TEST_ASSERT_EQUAL_INT(PATH_MAX, strlen(bigstr));
+	/*Checking cappability of reading not only visible sings*/
+	TEST_ASSERT_EQUAL_INT(43, strlen("\1\2\3\4\5\6\7\11\12\13\14\15\16\17\18\19\20\21\22\23\24\25\26\27\28\29\30\31\32\33\34\35\36\37\38\39\40"));
+	/*Adding null signs before actual value*/
+	TEST_ASSERT_EQUAL_INT(0, strlen("\0\0\0\0 1 \0\0 1 \0\0 1 \0"));
+	/*\3 and 99*/
+	TEST_ASSERT_EQUAL_INT(3, strlen("\399\0"));
```

Comment:
I think it does not check anything new, but we have to think about some pseudo random set of input data with different combinations of strings. It could be done in a separate test case and it may provide better coverage (we will discuss it)
```suggestion
```